src/main/java/org/opensearch/knn/index/util/Lucene.java

```diff
@@ -42,7 +42,7 @@ public class Lucene extends JVMLibrary {
                     )
                 )
                 .build()
-        ).addSpaces(SpaceType.L2, SpaceType.COSINESIMIL).build()
+        ).addSpaces(SpaceType.L2, SpaceType.COSINESIMIL, SpaceType.INNER_PRODUCT).build()
```

Comment:
nit: Should we move list of space types to a variable? This will be helpful if we start adding more space type.

Reply:
This is fine.. we are not adding space types for now. Plus this is how we have kept in other engines. 

https://github.com/opensearch-project/k-NN/blob/4734d88b8b12ff3edd676e3a1f337fc814c8932f/src/main/java/org/opensearch/knn/index/util/Nmslib.java#L50